pkg/zstdchunked/compression.go

```diff
@@ -0,0 +1,477 @@
+package zstdchunked
```

Comment:
Can we separate out this package into a separated module? From the perspective of the pkg consumer, it would be great if they can avoid having dependency to the entier c/storage project (contians drivers, etc.).

And based on experiment with zstd:chunked at , I propose to add prefetch support based on the logic of eStargz (https://github.com/containerd/stargz-snapshotter/blob/v0.4.1/docs/stargz-estargz.md).

Reply:
yes, I don't think it should be an issue.

@rhatdan what do you think?

Reply:
> And we should add prefetch support based on the logic of eStargz (https://github.com/containerd/stargz-snapshotter/blob/v0.4.1/docs/stargz-estargz.md) for mitigating the overhead of on-demand fetching .

could we add a probabilistic score to the TOC for each file?  e.g.

```
    {
      "name": "bin/busybox",
      "type": "reg",
      "size": 833104,
      "modtime": "2019-06-12T17:52:45Z",
      "mode": 33261,
      "offset": 126,
      "NumLink": 0,
      "digest": "sha256:8b7c559b8cccca0d30d01bc4b5dc944766208a53d18a03aa8afe97252207521f",
      "chunkDigest": "sha256:8b7c559b8cccca0d30d01bc4b5dc944766208a53d18a03aa8afe97252207521f",
     "accessScore: "1.00"
    },
```

so that the client can fine tune what files must be prefetched?

From your experience with estargz, how much is it important that these most frequently accessed files are contiguous in that tarball?

Reply:
Some registries doesn't support multiple Range Request so all prioritized files should be fetched by one range.

Reply:
thanks that makes sense.

My suggestion is to add an `accessScore` per file then have an optional pass that sorts the files in the tarball by their score.  In this way a client can decide the threshold to preload (e.g. `accessScore > 0.7`) then prefetch only the initial tarball part.
One disadvantage of such additional pass is that the diffID will be changed, so I think it should be optional so users can decide whether they want to keep the diffID or support registries that lack multiple Range Requests.

Reply:
SGTM

Reply:
I am all for breaking this into a separate REPO. Anything to make it easier to merge and move foreward.